homeassistant/components/google_drive/services.yaml

```diff
@@ -0,0 +1,24 @@
+append_sheet:
+  name: Append to Sheet
+  description: Append data to a worksheet in Google Sheets.
+  fields:
+    config_entry:
+      name: Config entry
+      description: Config entry to use.
+      required: true
+      selector:
+        config_entry:
+          integration: google_drive
+    worksheet:
+      name: Worksheet
+      description: Name of the worksheet. Defaults to the first one in the document.
+      example: "Sheet1"
+      selector:
+        text:
+    data:
```

Comment:
So here is an idea.

Dictionaries are sorted. So instead of having a time range etc, why don't we just accept a dictionary here.

And we map that to the columns in the first line of the sheet.

So let's say I pass:

```
data:
  hello: world
  cool: True
  count: 5
```

Then it will look for those columns in the sheet, and set the data to that row. For each key not mapped, add a new column.

That way we address @allenporter concern about offset, and dicts are easier than lists for users + everything is all organised.